### PR TITLE
oe/86bz12jgz/signup stepper

### PR DESCRIPTION
**Pull request by guardian-001**
## What does this do?

- Added stepper button
- Created signup stepper for architect
- Completed interface integration for the first step
- fixed back button logic
- fixed header title file 

## Why did you do this?

The changes were made to improve the architect onboarding experience by providing a clear and guided signup process. The stepper component enhances user interaction by visually indicating progress, which helps in reducing user confusion and drop-offs during the signup process.

## Who/what does this impact?

This impacts the signup interface for the architect

### CODE REVIEW

File: src/modules/architect/signup/demo-planning-confirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export default function DemoPlanningConfirmation({ onSubmit }: ResetFormProps) {
+  return (
+    <View className="flex h-fit items-center justify-between gap-16">
+      <View>
+        <Text
+          tx={'signupStep4.title'}
+          className="mb-2 text-center text-2xl font-extrabold"
+        />
+        <Text
+          tx={'signupStep4.description'}
```
Comment (MohamedMegdich1) [inline src/modules/architect/signup/demo-planning-confirmation.tsx:19]: signupStep4 avoid naming with number

File: src/modules/architect/signup/demo-planning-confirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export type ResetFormProps = {
+  onSubmit?: () => void;
+};
+
+export default function DemoPlanningConfirmation({ onSubmit }: ResetFormProps) {
+  return (
+    <View className="flex h-fit items-center justify-between gap-16">
+      <View>
+        <Text
+          tx={'signupStep4.title'}
```
Comment (MohamedMegdich1) [inline src/modules/architect/signup/demo-planning-confirmation.tsx:15]: the same

File: src/modules/architect/signup/demo-planning.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+export type ResetFormProps = {
+  onSubmit?: () => void;
+};
+
+export default function DemoPlanning({ onSubmit }: ResetFormProps) {
+  return (
+    <View className="flex h-fit items-center justify-between gap-16">
+      <View>
+        <Text
+          tx={'signupStep3.title'}
```
Comment (MohamedMegdich1) [inline src/modules/architect/signup/demo-planning.tsx:15]: the same

File: src/modules/architect/signup/demo-planning.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          className="mb-2 text-center text-2xl font-extrabold"
+        />
+        <Text
+          tx={'signupStep3.description'}
+          className="max-w-xs text-center text-sm text-description"
```
Comment (MohamedMegdich1) [inline src/modules/architect/signup/demo-planning.tsx:20]: the same

File: src/modules/architect/signup/signup-stepper.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  const handleNextStep = () => {
+    setStep(step + 1);
+  };
+  const stepsContent: {
+    title: TxKeyPath;
+    subtitle: TxKeyPath;
+    component: React.ReactNode;
+  }[] = [
+    {
+      title: 'signupStep1.title',
```
Comment (MohamedMegdich1) [inline src/modules/architect/signup/signup-stepper.tsx:33]: the same for all transaltion

File: src/shared/components/toggle-card.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import { View } from 'moti';
```
Comment (MohamedMegdich1) [inline src/shared/components/toggle-card.tsx:1]: why getting View from moti

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/architect/signup/demo-planning-confirmation.tsx (hunk 1 above)
Comment (guardian-001) [reply src/modules/architect/signup/demo-planning-confirmation.tsx:19]: fixed

File: src/modules/architect/signup/demo-planning-confirmation.tsx (hunk 2 above)
Comment (guardian-001) [reply src/modules/architect/signup/demo-planning-confirmation.tsx:15]: fixed

File: src/modules/architect/signup/demo-planning.tsx (hunk 3 above)
Comment (guardian-001) [reply src/modules/architect/signup/demo-planning.tsx:15]: fixed

File: src/modules/architect/signup/demo-planning.tsx (hunk 4 above)
Comment (guardian-001) [reply src/modules/architect/signup/demo-planning.tsx:20]: fixed

File: src/shared/components/toggle-card.tsx (hunk 6 above)
Comment (guardian-001) [reply src/shared/components/toggle-card.tsx:1]: fixed

File: src/modules/architect/signup/signup-stepper.tsx (hunk 5 above)
Comment (guardian-001) [reply src/modules/architect/signup/signup-stepper.tsx:33]: fixed

File: src/translations/fr/index.ts [hunk 7]
```diff
@@ -3,7 +3,12 @@ import login from './login.json';
 import onBoarding from './onboarding.json';
 import resetpass from './resetpass.json';
 import settings from './settings.json';
-import signupStep1 from './signup-step1.json';
+import signup from './signup.json';
+import signupStepSpeciality from './signup-step1.json';
+import signupStepCreateProfile from './signup-step2.json';
+import signupStepDemoPlanning from './signup-step3.json';
+import signupStepDemoPlanningConfirmation from './signup-step4.json';
+import signupStep5 from './signup-step5.json';
```
Comment (MohamedMegdich1) [inline src/translations/fr/index.ts:11]: signupStep5 change this also

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/translations/fr/index.ts (hunk 7 above)
Comment (guardian-001) [reply src/translations/fr/index.ts:11]: fixed

**Pull request merged**